src/dialogs/alignaudiodialog.cpp

```diff
@@ -0,0 +1,438 @@
+        emit progressUpdate(m_index, progress);
+    }
+
+signals:
+    void progressUpdate(int index, int percent);
+    void finished(int index, int offset, double drift, double quality);
+
+private:
+    AlignmentArray m_clipArray;
+    AlignmentArray& m_referenceArray;
+    AudioReader m_reader;
+    int m_index;
+    QFuture<void> m_future;
+    bool m_calculateDrift;
+};
+
+class AlignTableDelegate : public QStyledItemDelegate
+{
+    Q_OBJECT
+public:
+    void paint(QPainter *painter, const QStyleOptionViewItem &option, const QModelIndex &index) const
+    {
+        const AlignClipsModel* model = dynamic_cast<const AlignClipsModel*>(index.model());
+        switch (index.column()) {
+            case AlignClipsModel::COLUMN_ERROR:
+            {
+                QIcon icon;
+                if (!index.data().toString().isEmpty()) {
+                    icon = QIcon(":/icons/oxygen/32x32/status/task-reject.png");
+                } else if (model->getProgress(index.row()) == 100) {
+                    icon = QIcon(":/icons/oxygen/32x32/status/task-complete.png");
+                }
+                icon.paint(painter, option.rect, Qt::AlignCenter);
+                break;
+            }
+            case AlignClipsModel::COLUMN_NAME:
+            {
+                int progress = model->getProgress(index.row());
+                if (progress > 0 ) {
+                    QStyleOptionProgressBar progressBarOption;
+                    progressBarOption.rect = option.rect;
+                    progressBarOption.minimum = 0;
+                    progressBarOption.maximum = 100;
+                    progressBarOption.progress = progress;
+                    QApplication::style()->drawControl(QStyle::CE_ProgressBar, &progressBarOption, painter);
+                }
+                painter->drawText(option.rect, Qt::AlignLeft | Qt::AlignVCenter, index.data().toString() );
+                break;
+            }
+            case AlignClipsModel::COLUMN_OFFSET:
+//            case AlignClipsModel::COLUMN_DRIFT:
+                QStyledItemDelegate::paint(painter, option, index);
+                break;
+            default:
+                LOG_ERROR() << "Invalid Column" << index.row() << index.column();
+                break;
+        }
+    }
+};
+
+// Include this so that AlignTableDelegate can be declared in the source file.
+#include "alignaudiodialog.moc"
+
+AlignAudioDialog::AlignAudioDialog(QString title, MultitrackModel* model, const QVector<QUuid>& uuids, QWidget* parent)
+    : QDialog(parent)
+    , m_model(model)
+    , m_uuids(uuids)
+    , m_uiTask(nullptr)
+{
+    int col = 0;
+    setWindowTitle(title);
+    setWindowModality(QmlApplication::dialogModality());
+
+    QGridLayout* glayout = new QGridLayout();
+    glayout->setHorizontalSpacing(4);
+    glayout->setVerticalSpacing(2);
+    // Track Combo
+    glayout->addWidget(new QLabel(tr("Reference Audio Track")), col, 0, Qt::AlignRight);
```

Comment:
Only the first word in a field label is supposed to be capitalized: https://shotcut.org/notes/ui-conventions/

Reply:
done